From a2281eb74a1ac4d1ef1387493fc10f45fb5f74fd Mon Sep 17 00:00:00 2001
From: Ankit Gupta <agupta@cyngn.com>
Date: Tue, 5 Jul 2016 15:18:53 -0700
Subject: [PATCH 22/28] InputMethods: set the default value for preference

set the default value of the shared preference PREF_SHOW_SUGGESTIONS
directly from the xml file.

FEIJ-1483

Change-Id: If0e7c0411e8c26f926f16f5779c043f0675378fd
---
 java/res/xml/prefs_screen_correction.xml                   | 1 +
 .../latin/settings/CorrectionSettingsFragment.java         | 7 -------
 2 files changed, 1 insertion(+), 7 deletions(-)

diff --git a/java/res/xml/prefs_screen_correction.xml b/java/res/xml/prefs_screen_correction.xml
index f620ac05f..580bfd86b 100644
--- a/java/res/xml/prefs_screen_correction.xml
+++ b/java/res/xml/prefs_screen_correction.xml
@@ -51,6 +51,7 @@
         android:key="show_suggestions"
         android:summary="@string/prefs_show_suggestions_summary"
         android:title="@string/prefs_show_suggestions"
+        android:defaultValue="@bool/config_show_suggestions_default"
         android:persistent="true" />
     <CheckBoxPreference
         android:key="pref_key_use_personalized_dicts"
diff --git a/java/src/com/android/inputmethod/latin/settings/CorrectionSettingsFragment.java b/java/src/com/android/inputmethod/latin/settings/CorrectionSettingsFragment.java
index f3e7f6084..dfe899ece 100644
--- a/java/src/com/android/inputmethod/latin/settings/CorrectionSettingsFragment.java
+++ b/java/src/com/android/inputmethod/latin/settings/CorrectionSettingsFragment.java
@@ -90,13 +90,6 @@ public final class CorrectionSettingsFragment extends SubScreenFragment
 
         mUseContactsPreference = (SwitchPreference) findPreference(Settings.PREF_KEY_USE_CONTACTS_DICT);
         turnOffUseContactsIfNoPermission();
-
-        final SwitchPreference suggestionsEnabledPreference = (SwitchPreference) findPreference(
-                Settings.PREF_SHOW_SUGGESTIONS);
-        if (suggestionsEnabledPreference != null) {
-            suggestionsEnabledPreference.setChecked(
-                    Settings.getInstance().getCurrent().isSuggestionsEnabledPerUserSettings());
-        }
     }
 
     private void overwriteUserDictionaryPreference(final Preference userDictionaryPreference) {
-- 
2.17.1

